website/src/controllers/api-v3/members.js

```diff
@@ -231,4 +241,76 @@ api.getChallengeMemberProgress = {
   },
 };
 
+async function sendMessage (sender, userToReceiveMessage, messageData) {
+  let msg;
+
+  if (!messageData.type) {
+    msg = messageData.message;
+  } else {
+    msg = `Hello ${userToReceiveMessage.profile.name }, ${sender.profile.name} has sent you `;
+    msg += messageData.type === 'gems' ?  `${messageData.gems.amount} gems! ` : `${shared.content.subscriptionBlocks[messageData.subscription.key].months} months of subscription! `;
+    msg += messageData.message;
+  }
+
+  refPush(userToReceiveMessage.inbox.messages, chatDefaults(msg, sender));
+  userToReceiveMessage.inbox.newMessages++;
+  userToReceiveMessage._v++;
+  userToReceiveMessage.markModified('inbox.messages');
+
+  refPush(sender.inbox.messages, defaults({sent: true}, chatDefaults(msg, userToReceiveMessage)));
+  sender.markModified('inbox.messages');
+
+  let promises = [userToReceiveMessage.save(), sender.save()];
+  await Q.all(promises);
+}
+
+/**
+ * @api {posts} /send-private-message Get a challenge member progress
+ * @apiVersion 3.0.0
+ * @apiName SendPrivateMessage
+ * @apiGroup Members
+ *
+ * @apiParam {String} message The message
+ * @apiParam {UUID} toUserId The toUser _id
+ *
+ * @apiSuccess {} Object Returns an empty object
+ */
+api.sendPrivateMessage = {
+  method: 'POST',
+  url: '/send-private-message',
+  middlewares: [authWithHeaders(), cron],
+  async handler (req, res) {
+    req.checkBody('message', res.t('messageRequired')).notEmpty();
```

Comment:
Can anyone tell me how to return this custom message? I think testing and receiving these specific messages is crucial to a good API.